### PR TITLE
docs: use createResource instead of signal+effect

### PR DESCRIPTION
**Pull request by fredericoo**
## What kind of change does this PR introduce?

Replace bad practice way of fetching data in the example. Ideally It’d be a solidstart `server$` function, but for solid this is the way.

## What is the current behavior?

Using signal + effect, very react-inspired.

## What is the new behavior?

Using createResource, a much more ergonomic API and easier to understand.

### CODE REVIEW

File: apps/docs/pages/guides/getting-started/quickstarts/solidjs.mdx [hunk 1]
```diff
@@ -87,21 +87,17 @@ export const meta = {
...
       ```jsx src/App.jsx
         import { createClient } from "@supabase/supabase-js";
-        import { createEffect, createSignal, For } from "solid-js";
+        import { createResource, For } from "solid-js";
 
         const supabase = createClient('https://<project>.supabase.co', '<your-anon-key>');
 
+        async function getCountries() {
+          const { data } = await supabase.from("countries").select();
+          setCountries(data);
```
Comment (gregnr) [inline apps/docs/pages/guides/getting-started/quickstarts/solidjs.mdx:96]:
```suggestion
          return data;
```

Comment (fredericoo) [reply apps/docs/pages/guides/getting-started/quickstarts/solidjs.mdx]: wow went completely over my head, thanks for mentioning 😆

**Review by gregnr — APPROVED** [review]
Thanks @fredericoo!

**Comment by fredericoo** [comment]
@gregnr anything I can do for the vercel deploys to go through?

**Comment by gregnr** [comment]
Hey @fredericoo, apologies for the delay here. Looks like there were issues with the studio builds (unrelated to your changes). I've merged the latest `master` which should solve this. Will merge once builds pass.

**Pull request merged**